include/proxy/TransformInternal.h

```diff
@@ -49,6 +50,12 @@ class TransformTerminus : public VConnection
   int                   m_deletable;
   int                   m_closed;
   int                   m_called_user;
+
+private:
+  Event *_read_event     = nullptr;
+  bool   _read_disabled  = false;
+  Event *_write_event    = nullptr;
+  bool   _write_disabled = false;
```

Comment:
Is it safer for the disabled flags to default to true?

Reply:
That's a good point. My logic was to default this new "feature" I'm adding to off.

The initial values shouldn't make a difference, since the events we're protecting only happen during the do_io_* function calls which will explicitly set these to true or false, but I see your logic for defaulting the flags to disabled since the buffer will be nullptr. I'll run a few quick internal tests in production to verify I'm not missing something and update the PR if it seems fine.

Thanks for the review!

Reply:
Updated.